File: Sorting Algorithm/insertion_sort.py
# Insertion Sort Algorithm:

def InsertionSort(A,n):
    for i in range(1,n):
        value=A[i]
        hole=i
        while(hole>0 and A[hole-1]>value):
            A[hole]=A[hole-1]
            hole=hole-1
        A[hole]=value

if __name__=="__main__":
    print("Enter the elements in the array")
    A=list(map(int,input().split(" ")))
    n=len(A)
    InsertionSort(A,n)
    print("Sorted Array: ",end="")
    for i in range(0,n):
        print(A[i],end=" ")

'''
Time Complexity:
    1. Best Case: O(n)
    2. Average Case: O(n^2)
    3. Worst Case: O(n^2)

Space Complexity: O(1)
'''

File: Sorting Algorithm/InsertionSort.java
// Sorting array arr using insertion sort

public class InsertionSort{

     public static void main(String []args){
       int arr[] = {2,6,4,7,34,87,3};

        int len = arr.length;
        for(int i =1;i<len;i++)
        {
            int k = i-1;
            int element = arr[i];
            while(k>=0 && element<arr[k])
            {
                    arr[k+1] = arr[k];
                    arr[k] = element;
                    k--;
            }
        }
        for(int p =0; p<len;p++)
        {
            System.out.print(arr[p]+" ");
        }
     }
}
